tools/unit-tests/gen-test-fdlibm.c

```diff
@@ -0,0 +1,755 @@
+/* Copyright 2016 Samsung Electronics Co., Ltd.
+ * Copyright 2016 University of Szeged.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Unit test generator for fdlibm.
+ * To be compiled separately from the rest of jerry and to be linked to a trusted libm.
+ * Its output should be redirected to test-fdlibm.inc.h.
+ *
+ * Example:
+ * gcc gen-test-fdlibm.c -o gen-test-fdlibm -lm
+ * ./gen-test-fdlibm >test-fdlibm.inc.h
+ */
+
+#include <math.h>
+#include <stdio.h>
+
+#define GEN_INT_TEST(EXPR) printf("check_int (\"%s\", %s, %d);\n", #EXPR, #EXPR, EXPR);
+#define GEN_DBL_TEST(EXPR) printf("check_double (\"%s\", %s, %.20E);\n", #EXPR, #EXPR, EXPR);
+
+int
+main (int argc, char **args)
+{
+  printf ("/*\n"
+          " * Generated by tools/gen-test-fdlibm.sh\n"
+          " * DO NOT EDIT!!!\n"
+          " */\n");
+
+  /* acos tests */
+  GEN_DBL_TEST (acos (0.0));
```

Comment:
Neither-nor. At least, not exactly. I've been looking around for existing libm test suites but only found http://www.opensource.apple.com/source/Libm/Libm-40.11/noship.subproj/libm-test.c . It seemed to be shallow, however.

But it motivated me to test the well-known values (+-0.0, +-INF, NAN everywhere; PI, PI/2, PI/4, PI/6, ... angles for trigonometric functions; +-0.5, +-1.0 for arcus functions). And then I looked at fdlibm sources -- both the code and the doc comments -- and analyzed how the input domains are split up to ranges. I tried hard to add test inputs around both lower and upper bounds of all ranges. (E.g., `acos` is implemented to split the input domain at 0.5 -- among others --, thus I've added test inputs 0.4 and 0.6 to make sure that both ranges are tested.)

So, the values are manually chosen to test both normal and corner cases of fdlibm.